src/core/config/config.rs

```diff
@@ -123,6 +123,11 @@ pub struct Type {
     /// skip since it's set automatically by config transformer
     #[serde(skip_serializing)]
     pub key: Option<Key>,
+
+    ///
+    /// Any additional directives
+    #[serde(default, skip_serializing_if = "is_default")]
+    pub directives: Vec<Directive>,
```

Comment:
Define directives for Config separately. Config shouldn't depend on blueprint primitives ideally. Blueprint can depend on config though.

Reply:
I'm noticing some contradictory guidance based on your previous comment https://github.com/tailcallhq/tailcall/pull/2747#discussion_r1732148289. Could you please help clarify how to handle this so we can have a consistent approach moving forward? This clarification could also be valuable as part of the contribution guide for others. I believe we can avoid code duplication by reusing the implementation in one entity from another. What are your thoughts on this?

Reply:
moved this to core to reuse in both config and blueprint